tests/utils_test.ts

```diff
@@ -11,13 +11,27 @@ import {
   assertInstanceOf,
   assertMatch,
 } from "https://deno.land/std@0.170.0/testing/asserts.ts";
-import { _internals, buildUrl, execute, getSource } from "../src/utils.ts";
+import {
+  _internals,
+  buildRequestOptions,
+  execute,
+  getSource,
+} from "../src/utils.ts";
 import { RequestTimeoutError } from "../src/errors.ts";
+import { Config, config } from "../src/config.ts";
+import http from "node:http";
+import qs from "node:querystring";
 
 loadSync({ export: true });
-const BASE_URL = Deno.env.get("ENV_TYPE") === "local"
-  ? "http://localhost:3000"
-  : "https://serpapi.com";
+const BASE_OPTIONS = Deno.env.get("ENV_TYPE") === "local"
+  ? {
+    hostname: "localhost",
+    port: 3000,
+  }
+  : {
+    hostname: "serpapi.com",
+    port: 443,
+  };
```

Comment:
Let's include `protocol` here

```suggestion
  ? {
    hostname: "localhost",
    port: 3000,
    protocol: "http",
  }
  : {
    hostname: "serpapi.com",
    port: 443,
    protocol: "https",
  };
```